security-command-center/snippets/src/test/java/management/api/EventThreatDetectionCustomModuleTest.java

```diff
@@ -0,0 +1,166 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package management.api;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.securitycentermanagement.v1.EventThreatDetectionCustomModule;
+import com.google.cloud.securitycentermanagement.v1.ListEventThreatDetectionCustomModulesRequest;
+import com.google.cloud.securitycentermanagement.v1.SecurityCenterManagementClient;
+import com.google.cloud.securitycentermanagement.v1.SecurityCenterManagementClient.ListEventThreatDetectionCustomModulesPagedResponse;
+import com.google.cloud.testing.junit4.MultipleAttemptsRule;
+import com.google.common.base.Strings;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.StreamSupport;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class EventThreatDetectionCustomModuleTest {
+  // TODO(Developer): Replace the below variable
+  private static final String PROJECT_ID = System.getenv("SCC_PROJECT_ID");
+  private static final String CUSTOM_MODULE_DISPLAY_NAME = "java_sample_custom_module_test";
+  private static final int MAX_ATTEMPT_COUNT = 3;
+  private static final int INITIAL_BACKOFF_MILLIS = 120000; // 2 minutes
+
+  @Rule
+  public final MultipleAttemptsRule multipleAttemptsRule =
+      new MultipleAttemptsRule(MAX_ATTEMPT_COUNT, INITIAL_BACKOFF_MILLIS);
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setUp() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("SCC_PROJECT_ID");
+  }
+
+  @AfterClass
+  public static void cleanUp() throws IOException {
+    // Perform cleanup after running tests
+    cleanupExistingCustomModules();
+  }
+
+  // cleanupExistingCustomModules clean up all the existing custom module
+  private static void cleanupExistingCustomModules() throws IOException {
+    try (SecurityCenterManagementClient client = SecurityCenterManagementClient.create()) {
+      ListEventThreatDetectionCustomModulesRequest request =
+          ListEventThreatDetectionCustomModulesRequest.newBuilder()
+              .setParent(String.format("projects/%s/locations/global", PROJECT_ID))
+              .build();
+      ListEventThreatDetectionCustomModulesPagedResponse response =
+          client.listEventThreatDetectionCustomModules(request);
```

Comment:
please refactor to clean up only modules created by this collection of tests. do not delete "other" modules.

Reply:
Yes this condition I have already added in the code (if (module.getDisplayName().startsWith("java_sample_custom_module"))). It will be going to delete the modules created by this collection of tests and not the other modules

Reply:
Addressed, added etd prefix to the display name and it stands for event threat detection

Reply:
It does not help in the case this test is executed in parallel by multiple processes. For example, consider when this test is run in parallel in two environments with different JDKs.

Please using `java.util.UUID`. For the reference you can see how it is used in the [compute tests](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/05b16a2d97aad93cdac6fac4a27d8b85794758f1/compute/cloud-client/src/test/java/compute/InstanceOperationsIT.java#L79).

Reply:
Addressed, as suggested, now added the random UUID as a suffix to the display name